prov/rxm/src/rxm_init.c

```diff
@@ -109,7 +109,7 @@ int rxm_info_to_rxm(uint32_t version, const struct fi_info *core_info,
 		    struct fi_info *info)
 {
 	info->caps = rxm_info.caps;
-	info->mode = core_info->mode | rxm_info.mode;
```

Comment:
The core provider modes are needed here and below. RxM sets FI_LOCAL_MR/FI_MR_LOCAL mode bit based on whether the app supports it or not (see rxm_getinfo). Also, we would need to require OFI_MR_BASIC_MAP mode bits if the user had requested FI_RMA cap. This is handled by ofi_alter_domain_attr.

Reply:
Ok, it's clear.

But if a core provider requires `FI_LOCAL_MR/FI_MR_LOCAL` (like verbs/MSG), it reflect on RxM provider MR modes. i.e. `FI_LOCAL_MR/FI_MR_LOCAL` flag will be in fi_info's mode even if a user doesn't support it.

Reply:
No, the following code in rxm_getinfo should remove those modes if the user doesn't support it:
https://github.com/ofiwg/libfabric/blob/ca8dcaa85ce77cc436b03451f2fe5f8525665613/prov/rxm/src/rxm_init.c#L208

Reply:
@a-ilango you're my hero :) Thanks!
So, we need to remove the `FI_LOCAL_MR` in case of `FI_MR_BASIC` is specified and version is >=1.5.
Please, look at the proposed solution, it seems that it should be ok. It works fine for MPICH/CH4

Reply:
Ah, so there's an issue with current code. Thanks for the fix. This looks good.